app/models/product_import/entry_validator.rb

```diff
@@ -307,7 +309,9 @@ def mark_as_new_product(entry)
     end
 
     def mark_as_existing_variant(entry, existing_variant)
-      existing_variant.assign_attributes(entry.attributes.except('id', 'product_id'))
+      existing_variant.assign_attributes(
+        entry.attributes.except('id', 'product_id', 'shipping_category', 'tax_category')
+      )
```

Comment:
I get  "unknown attribute: producer" here: https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/branches/pull-request-4825/builds/17

Reply:
locally if I do:
```diff
diff --git a/app/models/product_import/entry_validator.rb b/app/models/product_import/entry_validator.rb
index f9b7ba8ab..d81997f39 100644
--- a/app/models/product_import/entry_validator.rb
+++ b/app/models/product_import/entry_validator.rb
@@ -62,8 +62,12 @@ module ProductImport
     end
 
     def mark_as_new_variant(entry, product_id)
+      puts entry.attributes
       new_variant = Spree::Variant.new(
-        entry.attributes.except('id', 'product_id', 'on_hand', 'on_demand', 'shipping_category', 'tax_category')
+        entry.attributes.except('id', 'product_id', 'on_hand', 'on_demand', 'shipping_category', 'tax_category',
+          'category', 'units', 'unit_type', 'variant_unit', 'variant_unit_scale', 'unscaled_units',
+          'enterprise', 'enterprise_id', 'producer', 'producer_id', 'primary_taxon_id'
+          )
       )
       new_variant.save
       new_variant.on_demand = entry.attributes['on_demand'] if entry.attributes['on_demand'].present?
@@ -310,7 +314,10 @@ module ProductImport
 
     def mark_as_existing_variant(entry, existing_variant)
       existing_variant.assign_attributes(
-        entry.attributes.except('id', 'product_id', 'shipping_category', 'tax_category')
+        entry.attributes.except('id', 'product_id',
+          'category', 'units', 'unit_type', 'variant_unit', 'variant_unit_scale', 'unscaled_units',
+          'enterprise', 'enterprise_id', 'producer', 'producer_id', 'primary_taxon_id',
+          'shipping_category', 'tax_category')
       )
       check_on_hand_nil(entry, existing_variant)

```
I get this one green: 
/dev/openfoodnetwork (3-0-latest) $ bin/rspec spec/models/product_importer_spec.rb:228

we probably need a better way of doing this... at least extract this list into a method...

Reply:
I think it just needs `producer` and not the others.

> we probably need a better way of doing this...

Agreed.

Reply:
I added only producer and the spec was still failing...

Reply:
I've updated it and it's now passing. I also found some dead code for a sub-feature that's not used anymore... :fire:

Reply:
great!
but we can define default in ProdImport, right?
It'd feel safer if that dead code was deleted and verified in master.

Reply:
Moved to #4856

Reply:
awesome!